seqr/views/apis/staff_api_tests.py

```diff
@@ -380,3 +640,31 @@ def test_success_story(self):
 
         self.assertEqual(len(response_json['rows']), 1)
         self.assertDictEqual(response_json['rows'][0], EXPECTED_SUCCESS_STORY)
+
+    @mock.patch('seqr.views.apis.staff_api.export_multiple_files')
+    @responses.activate
+    def test_anvil_export(self, mock_export_multiple_files):
+        url = reverse(anvil_export, args=[PROJECT_GUID])
+        _check_login(self, url)
+
+        # Create a SavedVariant row needed by this test
+        saved_variant = SavedVariant.objects.get(guid__exact = "SV0000001_2103343353_r0390_100")
+        variant_tag_type = VariantTagType.objects.get(name__exact = "Known gene for phenotype")
+        variant_tag = VariantTag.objects.create(variant_tag_type = variant_tag_type)
+        variant_tag.saved_variants.add(saved_variant)
+        variant_tag.save()
+
+        # We will test the inputs of the export_multiple_files method.
+        # Outputs of the method are not important for this test.
+        mock_export_multiple_files.return_value = HttpResponse("Dummy text", content_type="text/plain")
+
+        responses.add(responses.GET, '{}/Samples'.format(AIRTABLE_URL),
+                      json=AIRTABLE_SAMPLE_RECORDS, status=200)
+        responses.add(responses.GET, '{}/Collaborator'.format(AIRTABLE_URL),
+                      json=AIRTABLE_COLLABORATOR_RECORDS, status=200)
+        response = self.client.get(url)
+        self.assertEqual(response.status_code, 200)
+
+        mock_export_multiple_files.assert_called_with(EXPECTED_EXPORT_FILES,
+            u'1kg project n\u00e5me with uni\u00e7\u00f8de_AnVIL_Metadata',
+            add_header_prefix = True, file_format = 'tsv', blank_value = '-')
```

Comment:
Instead of specifying all the `EXPECTED_EXPORT_FILES` here, I would change this to `mock_export_multiple_files.assert_called_with(mock.ANY, u'1kg project n\u00e5me with uni\u00e7\u00f8de_AnVIL_Metadata',
            add_header_prefix = True, file_format = 'tsv', blank_value = '-')` which will test all the other parameters to the call. Then you can access what files it was called with using something like `exported_files = mock_export_multiple_files.call_args.args[0]` and you can do a series of test around that, like that it has the proper length and then for each file check it has the expected filename, header, and contains one correct row

Reply:
Good! Thank you for telling me about `call_args`. It is very useful in this case.